ome_zarr/io.py

```diff
@@ -0,0 +1,150 @@
+"""Reading logic for ome-zarr.
+
+Primary entry point is the :func:`~ome_zarr.io.parse_url` method.
+"""
+
+import json
+import logging
+import os
+import posixpath
+from abc import ABC, abstractmethod
+from typing import Optional
+from urllib.parse import urlparse
+
+import dask.array as da
+import requests
+
+from .types import JSONDict
+
+LOGGER = logging.getLogger("ome_zarr.io")
+
+
+class BaseZarrLocation(ABC):
+    """
+    Base IO primitive for reading Zarr data.
+
+    No assumptions about the existence of the given path string are made.
+    Attempts are made to load various metadata files and cache them internally.
+    """
+
+    def __init__(self, path: str) -> None:
+        self.zarr_path: str = path.endswith("/") and path or f"{path}/"
+        self.zarray: JSONDict = self.get_json(".zarray")
+        self.zgroup: JSONDict = self.get_json(".zgroup")
+        self.__metadata: JSONDict = {}
+        self.__exists: bool = True
+        if self.zgroup:
+            self.__metadata = self.get_json(".zattrs")
```

Comment:
not `.zgroup`?

Reply:
No. It's wanting the flexible metadata stuffs for multiscales, etc.

Reply:
:+1: Okay, just wanted to make sure given the avoidance of "or".